include/common/mir/logging/logger.h

```diff
@@ -58,6 +59,7 @@ class Logger
     virtual ~Logger() = default;
     Logger(const Logger&) = delete;
     Logger& operator=(const Logger&) = delete;
+    void format_message(std::ostream&, Severity, const std::string&, const std::string&);
```

Comment:
I don't think this interface is the correct place for formatting function that assumes streamed output.

This would not be needed, for example, by a logger that outputs to a database.

I would have no objection to a new `StreamingLogger` class (or a standalone function) to share implementation between `DumbConsoleLogger` and `FileLogger`.

Reply:
I didn't want to add new public API, but ACK. Will fix.

Reply:
This _is_ a new public API. ;)